src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -182,25 +176,33 @@ export const selectAccountTypeByAccountId = createSelector(
     accountMetadata[accountId]?.accountType,
 )
 
-export const selectIsPortfolioLoaded = createSelector(
-  selectAccountSpecifiers,
-  selectPortfolioAssetIds,
-  (accountSpecifiers, portfolioAssetIds) => {
-    if (!accountSpecifiers.length) return false
-    /**
-     * for a given wallet - we can support 1 to n chains
-     * AppContext ensures we will have a portfolioAssetId for each chain's fee asset
-     * until the portfolioAssetIds includes supported chains fee assets, it's not fully loaded
-     * the golf below ensures that's the case
-     */
-    const assetIdToChainId = (assetId: AssetId): ChainId => fromAssetId(assetId).chainId
-
-    return !size(
-      difference(
-        uniq(map(accountSpecifiers, flow([keys, head]))),
-        uniq(map(portfolioAssetIds, assetIdToChainId)),
-      ),
-    )
+type PortfolioState = 'loading' | 'success' | 'error'
+
+type PortfolioStateGranular = {
+  [k: AccountId]: PortfolioState
+}
+
+export const selectPortfolioStateGranular = createDeepEqualOutputSelector(
+  selectPortfolioAccountMetadata,
```

Comment:
account metadata is set by the app context when the wallet changes, keyed by `accountId`. these are the accounts request data from the backend.

portfolioAccounts is upserted by the portfolioSlice when the data comes in.

by comparing the existence of an account we can figure the loading state.

by comparing the assetIds belonging to that account, we can figure if we're in a successful or error state.